app/scripts/metamask-controller.js

```diff
@@ -760,6 +761,17 @@ export default class MetamaskController extends EventEmitter {
       state: initState.AnnouncementController,
     });
 
+    const networksOrderingMessenger = this.controllerMessenger.getRestricted({
+      name: 'NetworksOrderingController',
+    });
+    this.networksOrderingController = new NetworksOrderingController({
+      messenger: networksOrderingMessenger,
+      state: initState.NetworksOrderingController,
+      onNetworkStateChange: networkControllerMessenger.subscribe.bind(
```

Comment:
The controller messengers are intended just for use _inside_ the controller they're named for, e.g. the network controller messenger is only meant to be used inside the network controller. Here in the metamask controller, we should be using `this.controllerMessenger` for any subscriptions. This might explain why this wasn't working.

However as noted in a different comment, we can drop this event listener completely instead and rely just on the messaging system subscription in the constructior